podpac/core/data/types.py

```diff
@@ -694,7 +697,165 @@ def _find_h5py_keys(obj, keys=[]):
         keys = list(set(keys))
         keys.sort()
         return keys
-            
+
+class Zarr(DataSource):
+    source = tl.Unicode(allow_none=True)
+    group = tl.Instance(zarr.Group, allow_none=False)
+    datakey = tl.Unicode(allow_none=False).tag(attr=True)
+    latkey = tl.Unicode(allow_none=True, default_value='lat').tag(attr=True)
+    lonkey = tl.Unicode(allow_none=True, default_value='lon').tag(attr=True)
+    timekey = tl.Unicode(allow_none=True, default_value='time').tag(attr=True)
+    altkey = tl.Unicode(allow_none=True, default_value='alt').tag(attr=True)
+    dims = tl.List(trait=Dimension(), allow_none=False).tag(attr=True)
+    crs = tl.Unicode(allow_none=True, default_value=None).tag(attr=True)
+    cf_time = tl.Bool(False).tag(attr=True)
+    cf_units = tl.Unicode(allow_none=True).tag(attr=True)
+    cf_calendar = tl.Unicode(allow_none=True).tag(attr=True)
+
+    access_key_id = tl.Unicode()
+    secret_access_key = tl.Unicode()
+    region_name = tl.Unicode()
+
+    @tl.default('access_key_id')
+    def _get_access_key_id(self):
+        return settings['AWS_ACCESS_KEY_ID']
+
+    @tl.default('secret_access_key')
+    def _get_secret_access_key(self):
+        return settings['AWS_SECRET_ACCESS_KEY']
+
+    @tl.default('region_name')
+    def _get_region_name(self):
+        return settings['AWS_REGION_NAME']
+
+    def init(self):
+        # check that source or group is provided
+        if self.source is None:
+            self.group
```

Comment:
Does this force `_open_group` to be called?

Reply:
Yes. That's a light operation, and forces the Node to fail early (on init) rather than on eval. Actually, the subsequent checks use `self.group` anyways, so this isn't strictly necessary. I'd have no problem if you prefer to do less checks here and fail later.